proto/client-api/src/lib.rs

```diff
@@ -190,11 +194,53 @@ pub struct PeerId(pub u32);
 pub struct TrackId(pub u32);
 
 /// Credential used for a `Member` authentication.
-#[derive(
-    Clone, Debug, Deserialize, Display, Eq, From, Hash, PartialEq, Serialize,
-)]
-#[from(forward)]
-pub struct Credential(pub String);
+#[derive(Clone, Debug, Deserialize)]
+pub struct Credential(SecretString);
+
+impl Serialize for Credential {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: ser::Serializer,
+    {
+        self.0.expose_secret().serialize(serializer)
+    }
+}
+
+impl Credential {
+    /// Provides access to the underlying secret [`str`].
+    #[must_use]
+    pub fn expose_str(&self) -> &str {
+        self.0.expose_secret()
+    }
+}
+
+impl<T> From<T> for Credential
+where
+    T: ToString,
+{
+    fn from(value: T) -> Self {
+        Self(SecretString::new(value.to_string().into_boxed_str()))
```

Comment:
@evdokimovs using `ToString` here we allow leaking the secret in the memory without zeroizing it, because `.to_string()` accepts `&self` and clones the value.

So, imagine `value: String`. The `value.to_string()` produces another value in the heap, being then moved into `SecretString`, while the original `value` will die when the function ends, and gets deallocated on the heap without any zeroizing.

So here, we either require `T: ToString + Zeroize`, or require `T: Into<String>` and don't clone any values at all.

I'm leaning towards the seconf option.